site.py

```diff
@@ -0,0 +1,85 @@
+import os
+import subprocess
+import urllib
+import time
+import random
+from uuid import uuid4
+
+from flask import Flask, render_template, url_for, request, redirect, send_from_directory
+
+app = Flask(__name__)
+
+TEMP_DIR = 'tmp/'
+METADATA_FILE = 'metadata.csv'
+DATA_FILE = 'data.csv'
+
+# Retrieve the files uploaded by the user, and if they are valid,
+#  store them in a new generated subfolder of tmp/.
+@app.route('/', methods=['GET', 'POST'])
+def upload_file():
+    if request.method == 'GET':
+        return render_template('index.html')
+
+    if request.method == 'POST':
+        metadatafile = request.files['metadatafile']
+        datafile = request.files['datafile']
+        
+        if not metadatafile or not datafile:
+            render_template('index.html', missing_fields=True)
+
+        if not metadatafile.filename.endswith('.csv') \
+          or not datafile.filename.endswith('.csv'):
+            render_template('index.html', wrong_filetype=True)
+
+        print "Saving files..."
+        uniqueid = str(uuid4())
+        folder = os.path.join(TEMP_DIR, uniqueid)
+        os.mkdir(folder)
+
+        metadatafile.save(os.path.join(folder, METADATA_FILE))
+        datafile.save(os.path.join(folder, DATA_FILE))
+
+        return redirect(url_for('container', uuid=uniqueid))
+
+
+# Generate the Docker container from the uploaded files, 
+#  and redirect to the new Harvest instance.
+@app.route('/containers/<uuid>')
+def container(uuid):
+    folder = os.path.join(TEMP_DIR, uuid)
+    cidf = os.path.join(folder, 'docker.cid')
+    containerid = ''
+    if not os.path.exists(cidf):
+        print 'Files accepted. Generating Docker Container...' 
+        containerid = subprocess.check_output([
+                          'docker', 'run', '-d', '-P',
+                          '-v', (folder+':/input'),
+                          'dbhi/redcap-harvest'
+                      ])
+        containerid = containerid.strip()  # Get short container id for use in inspect
+        with open(cidf, 'w') as f:
+            f.write(containerid)
+    else:
+        with open(cidf) as f:
+            containerid = f.read()
+
+    portnumber = json.loads(subprocess.check_output(
+                                              ['docker','port', containerid, '8000']
+                                         ))
+   
+    print 'Waiting for harvest to load...' 
+    harvestIP = 'http://' + subprocess.check_output(['boot2docker', 'ip']).strip()\
```

Comment:
This won't work since boot2docker is not available inside a container nor does it apply to non-OS X systems. There does not seem to be a way to get the IP from the Docker daemon. So I think we need to pass this as environment parameter when this container runs.

```bash
docker run -v /var/run/docker.sock:/var/run/docker.sock -e DOCKER_HOST=`ip route|awk '/default/ { print  $3}'` dhbi/HaaS
```

Which can be used here as `os.environ.get('DOCKER_HOST')`. There may be more clever ways to do this, but this is simple enough to allows the using an IP or other hostname.